Mage/src/main/java/mage/abilities/keyword/ProtectionAbility.java

```diff
@@ -89,6 +91,10 @@ public boolean canTarget(MageObject source, Game game) {
                 return !((FilterCard) filter).match((Card) source, ((Permanent) source).getControllerId(), this, game);
             } else if (source instanceof Card) {
                 return !((FilterCard) filter).match((Card) source, ((Card) source).getOwnerId(), this, game);
+            } else if (source instanceof Token) {
+                // Fake a permanent with the Token info.
+                PermanentToken token = new PermanentToken((Token) source, null, game);
```

Comment:
@theelk801 @xenohedron So this is almost good enough. The only part not correct is that I do not have the player that will put the token into play.
So I'm afraid the protection from player cards are escaping some of the filtering due to using the 2 arguments `match`.

Reply:
I think this looks good to me but I'll leave it to @xenohedron to approve

Reply:
I think I'll still need to add something for protection from player down below.
Like If you [[Return Triumphant]] a [[True-Name Nemesis]], and somehow choose yourself for the protection.

Reply:
I think that can be done separately as it's something that matters for any aura token creation

Reply:
Alright, Yenna also has issues on that regard, but due to aura token using their raw target and not caring for protection. Need to think that through too.